aadarchi-gradle-plugin/src/main/java/io/github/Riduidel/aadarchi/AadarchiPlugin.java

```diff
@@ -0,0 +1,29 @@
+package io.github.Riduidel.aadarchi;
+
+import groovy.lang.Closure;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.ndx.aadarchi.base.ArchitectureDocumentationBuilder;
+
+import java.io.File;
+
+public class AadarchiPlugin implements Plugin<Project> {
+
+    @Override
+    public void apply(Project project) {
+        project.task("aadarchi", it -> it.doLast(new Closure<Void>(null) {
+            public Object doCall() throws Throwable {
+                System.setProperty("project.build.directory",
+                        String.join(
+                                File.separator,
+                                project.getProjectDir().getAbsolutePath(), "build"));
+                System.setProperty("project.basedir",
+                        String.join(
+                                File.separator,
+                                project.getProjectDir().getAbsolutePath()));

```

Comment:
That's ood, but there is nothing preventing people from using the whole set of maven properties. How will you have them handled in the future? Isn't there somewhere in the gradle universe a maven wrapping mechanism?

Reply:
This is not the maven issue. `All the aadarchi` are working as well when setting properties into gradle build.

But, code lines are using directly the maven properties into the `base` component. Therefore, we have to let the default value be usable even if we are in gradle.

This may be  mistake before but, may be fixed with some default property injector in `base` component which may know agnosticaly where is the working base directory